sdk/servicebus/service-bus/src/serviceBusAtomManagementClient.ts

```diff
@@ -464,18 +420,33 @@ export class ServiceBusAtomManagementClient extends ServiceClient {
    * @throws `RestError` with code that is a value from the standard set of HTTP status codes as documented at
    * https://docs.microsoft.com/en-us/dotnet/api/system.net.httpstatuscode?view=netframework-4.8
    */
-  async createQueue(queueName: string, queueOptions?: QueueOptions): Promise<CreateQueueResponse> {
-    log.httpAtomXml(
-      `Performing management operation - createQueue() for "${queueName}" with options: ${queueOptions}`
-    );
-    const response: HttpOperationResponse = await this.putResource(
-      queueName,
-      buildQueueOptions(queueOptions || {}),
-      this.queueResourceSerializer,
-      false
-    );
+  async createQueue(queue: QueueDescription): Promise<CreateQueueResponse>;
+  async createQueue(queueNameOrOptions: string | QueueDescription): Promise<CreateQueueResponse> {
+    if (typeof queueNameOrOptions == "string") {
```

Comment:
Consider using the if/else to build just the queue name and queue options. Rest of the code is the same for both blocks and need not be repeated. Same for other create* methods

Reply:
Okay, I'll refactor it.

Reply:
Refactored!